gobblin-metrics/src/main/java/gobblin/metrics/reporter/util/EventSubmitter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics.reporter.util;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Maps;
+
+import gobblin.metrics.GobblinTrackingEvent;
+import gobblin.metrics.MetricContext;
+
+
+/**
+ * Wrapper around Avro {@link gobblin.metrics.GobblinTrackingEvent.Builder} simplifying handling {@link gobblin.metrics.GobblinTrackingEvent}s.
+ *
+ * <p>
+ *   Instances of this class are immutable. Calling set* methods returns a copy of the calling instance.
+ * </p>
+ */
+public class EventSubmitter {
+
+  private final Map<String, String> metadata;
+  private final String namespace;
+  private final Optional<MetricContext> metricContext;
+
+  public static class Builder {
+    private final Optional<MetricContext> metricContext;
+    private final Map<String, String> metadata;
+    private final String namespace;
+
+    public Builder(MetricContext metricContext, String namespace) {
+      this(Optional.of(metricContext), namespace);
+    }
+
+    public Builder(Optional<MetricContext> metricContext, String namespace) {
+      this.metricContext = metricContext;
+      this.namespace = namespace;
+      this.metadata = Maps.newHashMap();
+    }
+
+    public Builder addMetadata(String key, String value) {
+      this.metadata.put(key, value);
+      return this;
+    }
+
+    public EventSubmitter build() {
+      return new EventSubmitter(this);
+    }
+  }
+
+  private EventSubmitter(Builder builder) {
+    this.metadata = builder.metadata;
+    this.namespace = builder.namespace;
+    this.metricContext = builder.metricContext;
+  }
+
+  /**
+   * Submits the {@link gobblin.metrics.GobblinTrackingEvent} to the {@link gobblin.metrics.MetricContext}.
+   * @param name Name of the event.
+   */
+  public void submit(String name) {
+    submit(name, Collections.<String, String>emptyMap());
```

Comment:
Please use `ImmutableMap.of()` in favor of `Collections.<String, String>emptyMap()`.